spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/properties/KafkaBinderConfigurationProperties.java

```diff
@@ -110,6 +110,12 @@ public Transaction getTransaction() {
 		return this.transaction;
 	}
 
+	/**
+	 * No longer used.
+	 * @return the connection String
+	 * @deprecated connection to zookeeper is no longer necessary
+	 */
+	@Deprecated
```

Comment:
Looks like you deprecate here a configuration property.

Please, consider this additions: https://docs.spring.io/spring-boot/docs/2.0.0.RELEASE/reference/htmlsingle/#configuration-metadata-property-attributes

> Deprecation can also be specified declaratively in code by adding the `@DeprecatedConfigurationProperty` annotation ...

```
@DeprecatedConfigurationProperty(replacement = "app.acme.name")
	@Deprecated
	public String getTarget() {
```

I think this is how it will appear in the meta-data and be suggested via IDE auto-completion assistant.

Reply:
As discussed on the standup - I think Oleg's intent is to just remove them during merge.

Reply:
Right, I don't mind. Just sharing the knowledge how properly deprecate configuration properties.